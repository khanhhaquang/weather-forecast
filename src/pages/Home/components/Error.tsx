import { useSelector } from 'react-redux';
import styled from 'styled-components';
import { getErrorMessage } from '../reducer';

const StyledWrapper = styled.span`
  color: red;
`;

const Error: React.FC<any> = () => {
  const message = useSelector(getErrorMessage);
  return message ? <StyledWrapper>{message}</StyledWrapper> : null;
};

export default Error;
