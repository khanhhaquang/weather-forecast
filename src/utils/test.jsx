/* eslint-disable import/no-unresolved */
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { render as rtlRender } from '@testing-library/react';
import rootReducer from 'pages/reducer';

function render(ui, { initialState, store = createStore(rootReducer, initialState) } = {}) {
  return {
    ...rtlRender(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}

export * from '@testing-library/react';
export { render };
