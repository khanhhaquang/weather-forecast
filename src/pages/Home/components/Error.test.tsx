import { render, screen } from 'utils/test';
import Error from './Error';

describe('Home/Error', () => {
  const mockState = {
    home: {
      error: {
        message: 'error',
      },
    },
  };

  it('should render correctly', () => {
    const { asFragment } = render(<Error />, { initialState: { ...mockState } });
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/error/i)).toBeInTheDocument();
  });
});
