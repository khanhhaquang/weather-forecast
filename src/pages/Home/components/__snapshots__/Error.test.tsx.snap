// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/Error should render correctly 1`] = `
<DocumentFragment>
  <span
    class="sc-bdnxRM jbauqE"
  >
    error
  </span>
</DocumentFragment>
`;
